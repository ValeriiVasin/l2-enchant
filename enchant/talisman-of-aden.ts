import { createEnchantMap } from '../helpers/create-enchant-map';
import { Item } from '../items';
import { EnchantStrategy } from '../types';

export const talismanOfAdenEnchantMap = createEnchantMap({
  strategy: EnchantStrategy.Destroy,
  chances: [100, 80, 70, 60, 40, 30, 24, 16, 10, 10],
  items: [
    Item.TalismanOfAden,
    Item.TalismanOfAden_1,
    Item.TalismanOfAden_2,
    Item.TalismanOfAden_3,
    Item.TalismanOfAden_4,
    Item.TalismanOfAden_5,
    Item.TalismanOfAden_6,
    Item.TalismanOfAden_7,
    Item.TalismanOfAden_8,
    Item.TalismanOfAden_9,
    Item.TalismanOfAden_10,
  ],
});
