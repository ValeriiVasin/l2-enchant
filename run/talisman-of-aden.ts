import { run } from '../helpers/run';
import { Item } from '../items';

const times = 10_000;

run(
  Item.TalismanOfAden,
  [
    Item.TalismanOfAden_1,
    Item.TalismanOfAden_2,
    Item.TalismanOfAden_3,
    Item.TalismanOfAden_4,
    Item.TalismanOfAden_5,
    Item.TalismanOfAden_6,
    Item.TalismanOfAden_7,
    Item.TalismanOfAden_8,
  ],
  times,
);
