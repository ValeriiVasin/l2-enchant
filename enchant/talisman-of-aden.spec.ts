import { talismanOfAdenEnchantMap } from './talisman-of-aden';

it('enchant map: talisman of aden', () => {
  expect(talismanOfAdenEnchantMap).toMatchSnapshot();
});
