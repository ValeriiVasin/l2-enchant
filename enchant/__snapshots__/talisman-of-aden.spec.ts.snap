// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: talisman of aden 1`] = `
Map {
  "Талисман Адена" => Object {
    "cost": "Nothing",
    "item": "Талисман Адена",
    "results": Array [
      Object {
        "chance": 100,
        "item": "Талисман Адена +1",
      },
      Object {
        "chance": 0,
        "item": "Nothing",
      },
    ],
  },
  "Талисман Адена +1" => Object {
    "cost": "Nothing",
    "item": "Талисман Адена +1",
    "results": Array [
      Object {
        "chance": 80,
        "item": "Талисман Адена +2",
      },
      Object {
        "chance": 20,
        "item": "Nothing",
      },
    ],
  },
  "Талисман Адена +2" => Object {
    "cost": "Nothing",
    "item": "Талисман Адена +2",
    "results": Array [
      Object {
        "chance": 70,
        "item": "Талисман Адена +3",
      },
      Object {
        "chance": 30,
        "item": "Nothing",
      },
    ],
  },
  "Талисман Адена +3" => Object {
    "cost": "Nothing",
    "item": "Талисман Адена +3",
    "results": Array [
      Object {
        "chance": 60,
        "item": "Талисман Адена +4",
      },
      Object {
        "chance": 40,
        "item": "Nothing",
      },
    ],
  },
  "Талисман Адена +4" => Object {
    "cost": "Nothing",
    "item": "Талисман Адена +4",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Талисман Адена +5",
      },
      Object {
        "chance": 60,
        "item": "Nothing",
      },
    ],
  },
  "Талисман Адена +5" => Object {
    "cost": "Nothing",
    "item": "Талисман Адена +5",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Талисман Адена +6",
      },
      Object {
        "chance": 70,
        "item": "Nothing",
      },
    ],
  },
  "Талисман Адена +6" => Object {
    "cost": "Nothing",
    "item": "Талисман Адена +6",
    "results": Array [
      Object {
        "chance": 24,
        "item": "Талисман Адена +7",
      },
      Object {
        "chance": 76,
        "item": "Nothing",
      },
    ],
  },
  "Талисман Адена +7" => Object {
    "cost": "Nothing",
    "item": "Талисман Адена +7",
    "results": Array [
      Object {
        "chance": 16,
        "item": "Талисман Адена +8",
      },
      Object {
        "chance": 84,
        "item": "Nothing",
      },
    ],
  },
  "Талисман Адена +8" => Object {
    "cost": "Nothing",
    "item": "Талисман Адена +8",
    "results": Array [
      Object {
        "chance": 10,
        "item": "Талисман Адена +9",
      },
      Object {
        "chance": 90,
        "item": "Nothing",
      },
    ],
  },
  "Талисман Адена +9" => Object {
    "cost": "Nothing",
    "item": "Талисман Адена +9",
    "results": Array [
      Object {
        "chance": 10,
        "item": "Талисман Адена +10",
      },
      Object {
        "chance": 90,
        "item": "Nothing",
      },
    ],
  },
}
`;
